src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -21,148 +87,1634 @@ const ActorInit En_Ossan_InitVars = {
+    static Vec3f sSelectedItemPosition[] = {
+        { 35.0f, 68.0f, -130.0f },
+        { -35.0f, 68.0f, -130.0f },
+    };
+
+    EnGirlA* item;
+    u8 i;
+    u8 i2;
+    ShopItem* shopItem;
+    Vec3f worldPos;
+
+    i = this->cursorIdx;
+    shopItem = &sShops[this->actor.params][i];
+    item = this->items[i];
+
+    i2 = i >> 2;
+    VEC3F_LERPIMPDST(&worldPos, &shopItem->spawnPos, &sSelectedItemPosition[i2], this->shopItemSelectedTween);
+
+    item->actor.world.pos.x = worldPos.x;
+    item->actor.world.pos.y = worldPos.y;
+    item->actor.world.pos.z = worldPos.z;
+}
+
+void EnOssan_ResetItemPosition(EnOssan* this) {
+    this->shopItemSelectedTween = 0.0f;
+    EnOssan_PositionSelectedItem(this);
+}
+
+/*
+ * Returns true if animation has completed
+ */
+s32 EnOssan_TakeItemOffShelf(EnOssan* this) {
+    Math_ApproachF(&this->shopItemSelectedTween, 1.0f, 1.0f, 0.15f);
+    if (this->shopItemSelectedTween >= 0.85f) {
+        this->shopItemSelectedTween = 1.0f;
+    }
+    EnOssan_PositionSelectedItem(this);
+    if (this->shopItemSelectedTween == 1.0f) {
+        return true;
+    }
+    return false;
+}
+
+/*
+ * Returns true if animation has completed
+ */
+s32 EnOssan_ReturnItemToShelf(EnOssan* this) {
+    Math_ApproachF(&this->shopItemSelectedTween, 0.0f, 1.0f, 0.15f);
+    if (this->shopItemSelectedTween <= 0.15f) {
+        this->shopItemSelectedTween = 0.0f;
+    }
+    EnOssan_PositionSelectedItem(this);
+    if (this->shopItemSelectedTween == 0.0f) {
+        return true;
+    }
+    return false;
+}
+
+void EnOssan_UpdateItemSelectedProperty(EnOssan* this) {
+    EnGirlA** items = this->items;
+    EnGirlA* item;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(this->items); i++, items++) {
+        item = *items;
+        if (item != NULL) {
+            if (this->actionFunc != EnOssan_SelectItem && this->actionFunc != EnOssan_CannotBuy &&
+                this->drawCursor == 0) {
+                item->isSelected = false;
+            } else {
+                item->isSelected = this->cursorIdx == i ? true : false;
+            }
+        }
+    }
+}
+
+void EnOssan_UpdateCursorAnim(EnOssan* this) {
+    f32 t;
+
+    t = this->cursorAnimTween;
+    if (this->cursorAnimState == 0) {
+        t += 0.05f;
+        if (t >= 1.0f) {
+            t = 1.0f;
+            this->cursorAnimState = 1;
+        }
+    } else {
+        t -= 0.05f;
+        if (t <= 0.0f) {
+            t = 0.0f;
+            this->cursorAnimState = 0;
+        }
+    }
+    this->cursorColorR = COL_CHAN_MIX(0, 0.0f, t);
+    this->cursorColorG = COL_CHAN_MIX(80, 80.0f, t);
+    this->cursorColorB = COL_CHAN_MIX(255, 0.0f, t);
+    this->cursorColorA = COL_CHAN_MIX(255, 0.0f, t);
+    this->cursorAnimTween = t;
+}
+
+void EnOssan_UpdateStickDirectionPromptAnim(EnOssan* this) {
+    f32 arrowAnimTween = this->arrowAnimTween;
+    f32 stickAnimTween = this->stickAnimTween;
+    s32 new_var2 = 255;
+    f32 new_var3;
```

Comment:
This function looks like it was a pain, but it would be nice if these were named a little more descriptively.

Reply:
I'm not sure what to rename `new_var3` to. I'm 99.9% certain its fake, but could not find a way to remove it.